### PR TITLE
Enhancing AddPaperModal

### LINKED ISSUE #28: 'Adding Papers to My Collections' Pop-up
To bridge the gap between papers and collections, we need to implement pop-up where users can add papers to their own collections. The appearance of it would look like this.
[Add Papers to Collections Pop-up(2_4)](https://github.com/swsnu/swpp2019-team3/wiki/Requirements-and-Specification-Document#2_4-save-1)

### PR DESCRIPTION
**Pull request by davin111**
Related Issue: #28 
This PR will resolve #28 issue.

# Major changes
## Backend
GET /api/collection/user API input and output changed.
- input: As before it was, the input format can be `{id: user_id}`. However, if needed, `{id: user_id, paper: paper_id}` is also possible.
- output: When `paper` key is given as the input, each collection dict contains boolean value(`contains_paper: true/false`) which indicates whether the collection contains the given paper. If `paper` key is not given, the value of `contains_paper` is always false.

## Frontend
- Checkboxes of AddPaperModal reflects whether the user's collections contain the given paper.
스크린샷 2019-11-12 03 30 07
- if users try to click the add button when nothing changed, "Nothing changed!" message is shown.
스크린샷 2019-11-12 03 30 10
- Because users actually not only adding to collections but also removing from collections on the AddPaperModal, the message after changing was modified appropriately.
스크린샷 2019-11-12 03 30 46
- Now, 'GET /api/collection/user API' is called not when the component is mounted, but when the button opening AddPaperModal is clicked.

# Minor changes
- When an exception is raised, views.py print the stringfied exception.
- renamed 'Collection' dir in actions and reducers to 'collection'.
- removed onGetMe and related codes from AddPaperModal
- Related tests were implemented. Sorry for some missing tests about AddPaperModal because of async problems. :(

### Checklist
This pull request...
- [x] has preceding issues related with it (if resolves some issues, metions them with 'resolve')
- [x] doesn't include too many changes (you should focus on the specific feature in a PR)
- [x] elaborately explains what feature is added, which part is fixed or improved
- [x] has step by step instructions for collaborators, if needed
- [x] has attached screenshots, if appropriate

My codes...
- [ ] include enough tests for added and changed parts
- [x] passe all the new and existing tests
- [x] include appropriate comments
- [x] don't introduce unnecessary warnings

### CODE REVIEW

File: frontend/src/store/actions/collection/collection.js [hunk 1]
```diff
@@ -35,12 +35,13 @@ const getCollectionsByUserIdFailure = (error) => ({
     target: error,
 });
 
-export const getCollectionsByUserId = (userId) => (dispatch) => axios.get("api/collection/user", { params: userId })
+// params can be { id: user_id } or { id: user_id, paper: id }
+export const getCollectionsByUserId = (params) => (dispatch) => axios.get("/api/collection/user", { params })
```
Comment (kosumin) [inline frontend/src/store/actions/collection/collection.js]: Is this working well without { params: { id: 1 } } ?

Comment (davin111) [reply frontend/src/store/actions/collection/collection.js]: @kosumin Yeah, actually, Eslint automatically changed it.
Also as I mentioned on the explanation and the comment of that code, now you can give the format { id: user_id, paper: id } as an input to the action.

Comment (kosumin) [reply frontend/src/store/actions/collection/collection.js]: Good! Thank you

**Comment by abcthing** [comment]
LGTM!

**Pull request merged**